src/aggregate.js

```diff
@@ -38,31 +42,24 @@ const config = require('./config');
  */
 module.exports = async function aggregate(collection, params) {
   params = _.defaults(await sanitizeParams(collection, params), { aggregation: [] });
-  const cursorQuery = generateCursorQuery(params);
-  const $sort = generateSort(params);
 
-  let index = _.findIndex(params.aggregation, (step) => !_.isEmpty(step.$match));
+  const $match = generateCursorQuery(params);
+  const $sort = generateSort(params);
+  const $limit = params.limit + 1;
 
-  if (index < 0) {
-    params.aggregation.unshift({ $match: cursorQuery });
-    index = 0;
+  let sortSteps;
+  if (params.sortCaseInsensitive) {
+    sortSteps = [{ addFields: { __lc: { $toLower: '$' + params.paginatedField } } }, { $sort }];
```

Comment:
With that approach, a custom $sort won't be supported. I mean, the user will be able to execute it at the beginning of the pipeline, but this one will prevail.

Reply:
Any sorting different from the pagination criteria would result in pagination issues: first because the user would get confused as what the pages contain; second because the `next` and `previous` handles are obtained from the first and last document in the collection, so if these are not in pagination order we would end up with overlapping pages.

Reply:
We should throw an error if the user is trying to custom $sort then. At least to warn the user that $sort is not supported.

Reply:
Sorting is safe if we sort again by the pagination field afterwards. We have such an example in our codebase: we sort by timestamp, then group to take the last one of each name, then page the results (requiring sorting again, of course).

Reply:
Oh ok. So, the user could also use a virtual `paginatedField` right? E.g.

```
const res = await paging.aggregate(collection, {
  aggregation: [
    { $group: { _id: '$fieldA', count: { $sum: 1 } } }
  ],
  paginatedField: 'count',
  sortAscending: false,
  limit: 3,
});
```

To my understanding, this is also a new capability unlocked by this PR.

Reply:
Yes. That's also possible.